src/arch/riscv/tlb.cc

```diff
@@ -418,9 +425,20 @@ TLB::translate(const RequestPtr &req, ThreadContext *tc,
 
         Process * p = tc->getProcessPtr();
 
-        Fault fault = p->pTable->translate(req);
-        if (fault != NoFault)
-            return fault;
+        /*
+         * In RV32, as vaddr >= 0x80000000 is legal in userspace (except for
```

Comment:
Is this part of the ISA spec or is it Linux specific?

Reply:
Oh. Indeed, it's Linux-specific. I will fix it soon.